### PR TITLE
Support Edge Gateway DHCP Forwarding endpoint

### PR DESCRIPTION
**Pull request by adezxc**
## Description

This PR adds Edge Gateway DHCP Forwarding support with two new methods `UpdateDhcpForwarder` and `GetDhcpForwarder`, supported in VCD versions v10.3.1+

### CODE REVIEW

File: govcd/nsxt_edgegateway_test.go [hunk 1]
```diff
@@ -502,3 +502,66 @@ func (vcd *TestVCD) Test_NsxtEdgeQoS(check *C) {
...
+	// Check that updates were applied
+	check.Assert(updatedEdgeDhcpForwarderConfig.Enabled, Equals, true)
+	check.Assert(updatedEdgeDhcpForwarderConfig.DhcpServers, DeepEquals, testDhcpServers)
+
+	disabledDhcpForwarderConfig := &types.NsxtEdgeGatewayDhcpForwarder{
+		Enabled:     false,
+		DhcpServers: testDhcpServers,
+	}
+
+	// Update DHCP forwarder config
```
Comment (dataclouder) [inline govcd/nsxt_edgegateway_test.go:559]: If I understand correctly, we are testing the change in the `Enabled` property, but not the removal or addition of any DHCP servers.
Should we also test that?

File: govcd/nsxt_edgegateway.go [hunk 2]
```diff
@@ -658,6 +658,61 @@ func (egw *NsxtEdgeGateway) UpdateQoS(qosConfig *types.NsxtEdgeGatewayQos) (*typ
...
+		return nil, err
+	}
+
+	return dhcpForwarder, nil
+}
+
+// UpdateDhcpForwarder updates DHCP forwarder configuration for an NSX-T Edge Gateway
+func (egw *NsxtEdgeGateway) UpdateDhcpForwarder(dhcpForwarderConfig *types.NsxtEdgeGatewayDhcpForwarder) (*types.NsxtEdgeGatewayDhcpForwarder, error) {
+	if egw.EdgeGateway == nil || egw.client == nil || egw.EdgeGateway.ID == "" {
+		return nil, fmt.Errorf("cannot get DHCP forwarder for NSX-T Edge Gateway without ID")
```
Comment (Didainius) [inline govcd/nsxt_edgegateway.go:691]:
```suggestion
		return nil, fmt.Errorf("cannot update DHCP forwarder for NSX-T Edge Gateway without ID")
```

File: types/v56/nsxt_types.go [hunk 3]
```diff
@@ -1684,6 +1684,13 @@ type NsxtEdgeGatewayQos struct {
 	IngressProfile *OpenApiReference `json:"ingressProfile"`
 }
 
+// NsxtEdgeGatewayDhcpForwarder provides DHCP forwarding configuration on an Edge Gateway by defining
+// DHCP servers
+type NsxtEdgeGatewayDhcpForwarder struct {
+	Enabled     bool     `json:"enabled"`
+	DhcpServers []string `json:"dhcpServers"`
+}
```
Comment (lvirbalas) [inline types/v56/nsxt_types.go:1692]: Just a question: what is your verdict for the version field?

**Review by lvirbalas — APPROVED** [review]

File: govcd/nsxt_edgegateway_test.go (hunk 1 above)
Comment (adezxc) [reply govcd/nsxt_edgegateway_test.go:581]: Agreed, updated now.

**Review by Didainius — APPROVED** [review]

**Review by adambarreiro — APPROVED** [review]

**Review by dataclouder — APPROVED** [review]

**Pull request merged**